src/url-replacements.js

```diff
@@ -94,6 +97,38 @@ class UrlReplacements {
             Promise.resolve());
       });
     });
+
+    this.set_('TIMESTAMP', () => {
+      return new Date().getTime();
+    });
+
+    this.set_('TIMEZONE', () => {
+      return new Date().getTimezoneOffset();
+    });
+
+    this.set_('SCROLL_TOP', () => {
+      return viewportFor(this.win_).getScrollTop();
+    });
+
+    this.set_('SCROLL_LEFT', () => {
+      return viewportFor(this.win_).getScrollLeft();
+    });
+
+    this.set_('SCROLL_WIDTH', () => {
+      return viewportFor(this.win_).getScrollWidth();
```

Comment:
One possible problem: we currently have some cases where we can't reliably tell the scroll width. This is due to numerous iOS issues with iframes.